src/core/tsi/s2a/record_protocol/s2a_crypter.cc

```diff
@@ -55,6 +56,23 @@ typedef struct s2a_crypter {
       : tls_version(version), ciphersuite(cipher) {}
 } s2a_crypter;
 
+/** This function returns the tag size of the ciphersuite supported by
+ *  |crypter|. The caller must not pass in nullptr for |crypter|.
+ *  - crypter: an instance of s2a_crypter. **/
+static size_t s2a_tag_size(const s2a_crypter* crypter) {
+  GPR_ASSERT(crypter != nullptr);
+  switch (crypter->ciphersuite) {
+    case kTlsAes128GcmSha256:
+    case kTlsAes256GcmSha384:
+      return kEvpAeadAesGcmTagLength;
+    case kTlsChacha20Poly1305Sha256:
+      return kPoly1305TagLength;
+    default:
+      gpr_log(GPR_ERROR, kS2AUnsupportedCiphersuite);
+      abort();
```

Comment:
`abort()` seems a little strong to me. Can we handle it more gracefully?

Reply:
In the most recent commit (to come), I refactored this method to output a status code and populate some error details in this case.